server/app/views/category.py

```
@@ -0,0 +1,104 @@
+    return PaginatedCategories(
+        data=categories,
+        total=total,
+        page=(skip // limit) + 1,
+        page_size=limit,
+        total_pages=total_pages,
+    )
+
+
+@router.get("/{category_number}", response_model=Category, operation_id="getCategory")
+async def get_category(
+    category_number: int, repo: CategoryRepository = Depends(get_category_repository)
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)